classes/class-p4-master-site.php

```diff
@@ -605,6 +614,59 @@ public function view_meta_box_search( $post ) {
 		<?php
 	}
 
+	/**
+	 * Forces a user to login using Google Auth if they have a greenpeace.org email
+	 *
+	 * @param WP_User $user The current user logging in.
+	 * @param String  $username The username of the user.
+	 * @param String  $password The password of the user.
+	 */
+	public function enforce_google_signon( $user, $username = null, $password = null ) {
```

Comment:
Nitpicking about type hints, but it helps to catch error early, for both functions
```php
	 * @param null|WP_User|WP_Error $user The current user logging in.
	 * @param null|string           $username The username of the user.
	 * @param null|string           $password The password of the user.
	 * @return null|WP_User|WP_Error
	 */
	public function ... ( $user, ?string $username = null, ?string $password = null ) {
```

Reply:
Nice!